Modern/ops/src/main/java/org/bonej/ops/ellipsoid/FindEllipsoidFromBoundaryPoints.java

```diff
@@ -0,0 +1,343 @@
+ * </p>
+ *
+ * @author Alessandro Felder
+ */
+@Plugin(type = Op.class)
+public class FindEllipsoidFromBoundaryPoints extends AbstractBinaryFunctionOp<List<ValuePair<Vector3d,Vector3d>>,Vector3d,Optional<Ellipsoid>> {
+
+    private static QuadricToEllipsoid quadricToEllipsoid = new QuadricToEllipsoid();
+    private static EllipsoidPlaneIntersection intersectionOp = new EllipsoidPlaneIntersection();
+
+    @Parameter(persist=false, required = false)
+    private double calibratedLargestImageDimension = 100.0;
+
+    @Override
+    public Optional<Ellipsoid>calculate(List<ValuePair<Vector3d,Vector3d>> fourVerticesWithNormals, Vector3d centre) {
+        List<Vector3d> fourVertices = new ArrayList<>();
+        fourVerticesWithNormals.forEach(v -> fourVertices.add(v.getA()));
+
+        fourVerticesWithNormals.sort(Comparator.comparingDouble(p->euclideanDistance(p.getA(),centre)));
+        VertexWithNormal p = new VertexWithNormal(fourVerticesWithNormals.get(0));
+        fourVertices.remove(p.getVertex());
+
+        ValuePair<VertexWithNormal, Double> qAndRadius = calculateQ(fourVertices, p);
+        if (qAndRadius == null || qAndRadius.getB()>calibratedLargestImageDimension) return Optional.empty();
+        fourVertices.remove(qAndRadius.getA().getVertex());
+
+        Vector3d np = new Vector3d(p.getNormal());
+        np.mul(qAndRadius.getB());
+        p.setNormal(np);
+
+        Vector3d c = new Vector3d(p.getNormal());
+        c.add(p.getVertex());
+        Matrix4d q1 = getQ1(c, qAndRadius.getB());
+        Matrix4d q2 = getQ2(p, qAndRadius.getA());
+
+        ValuePair<Vector3d, Double> rAndAlpha = calculateSurfacePointAndGreekCoefficient(q1, q2, fourVertices);
+        if (rAndAlpha == null) return Optional.empty();
+        fourVertices.remove(rAndAlpha.getA());
+
+        Matrix4d q1PlusAlphaQ2 = new Matrix4d(q2);
+        Matrix4d fullScalingMatrix = new Matrix4d();
+        fullScalingMatrix.scaling(rAndAlpha.getB());
+        fullScalingMatrix.m33(rAndAlpha.getB());
+        q1PlusAlphaQ2.mul(fullScalingMatrix, q1PlusAlphaQ2);
+        q1PlusAlphaQ2.add(q1, q1PlusAlphaQ2);
+
+        Optional<Ellipsoid> ellipsoid = quadricToEllipsoid.calculate(q1PlusAlphaQ2);
+        if(!ellipsoid.isPresent()) return ellipsoid;
+
+        Matrix4d q3 = getQ3(Arrays.asList(p.getVertex(), qAndRadius.getA().getVertex(), rAndAlpha.getA()), ellipsoid.get());
+
+        ValuePair<Vector3d, Double> sAndBeta = calculateSurfacePointAndGreekCoefficient(q1PlusAlphaQ2, q3, fourVertices);
+        if (sAndBeta == null) return ellipsoid;
+
+        Matrix4d q1PlusAlphaQ2plusBetaQ3 = new Matrix4d(q3);
+        fullScalingMatrix.scaling(sAndBeta.getB());
+        fullScalingMatrix.m33(sAndBeta.getB());
+        q1PlusAlphaQ2plusBetaQ3.mul(fullScalingMatrix, q1PlusAlphaQ2plusBetaQ3);
+        q1PlusAlphaQ2plusBetaQ3.add(q1PlusAlphaQ2);
+
+        return quadricToEllipsoid.calculate(q1PlusAlphaQ2plusBetaQ3);
```

Comment:
Similar helper method `createEllipsoid()` here. Maybe even same method? There seems to be more or less duplicated code.